### PR TITLE
Samuela/jupyter stuff

### PR DESCRIPTION
**Pull request by samuela**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/nbclient/default.nix [hunk 1]
```diff
@@ -1,27 +1,65 @@
...
+
   disabled = pythonOlder "3.7";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-QMUsm148MfrsruafICs/U+ONfBxWPeD63enX7aD9r+g=";
+  src = fetchFromGitHub {
+    owner = "jupyter";
+    repo = pname;
+    rev = "refs/tags/v${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nbclient/default.nix:30]:
```suggestion
    rev = "v${version}";
```

File: pkgs/development/python-modules/nbclient/default.nix [hunk 2]
```diff
@@ -1,27 +1,65 @@
...
+, traitlets
+, xmltodict
 }:
 
 buildPythonPackage rec {
   pname = "nbclient";
-  version = "0.5.13";
+  version = "0.6.2";
+  # As of version 0.6.2, there is both a setup.py and pyproject.toml. However,
+  # the pyproject.toml does not appear to be the main entry point.
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nbclient/default.nix:22]:
```suggestion
```
pyproject.toml can be used for linting tool settings. I don't think we need a comment for that each time.

File: pkgs/development/python-modules/ipywidgets/default.nix [hunk 3]
```diff
@@ -1,48 +1,40 @@
...
 
-  checkPhase = ''
-    ${python.interpreter} -m unittest discover
-  '';
+  checkInputs = [ pytestCheckHook ];
 
-  meta = {
+  meta = with lib; {
     description = "IPython HTML widgets for Jupyter";
     homepage = "http://ipython.org/";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ipywidgets/default.nix:36]:
```suggestion
    homepage = "https://ipython.org/";
```

File: pkgs/development/python-modules/nbconvert/default.nix [hunk 4]
```diff
@@ -41,46 +44,55 @@ buildPythonPackage rec {
     substituteAllInPlace ./nbconvert/exporters/templateexporter.py
   '';
 
-  checkInputs = [ pytestCheckHook glibcLocales ];
-
   propagatedBuildInputs = [
-    entrypoints bleach mistune jinja2 pygments traitlets testpath
-    jupyter_core nbformat ipykernel pandocfilters tornado jupyter-client
-    defusedxml beautifulsoup4
     (nbclient.override { doCheck = false; }) # avoid infinite recursion
```
[the remark this answers is not in the document]
Comment (FRidh) [reply pkgs/development/python-modules/nbconvert/default.nix:48]: https://github.com/NixOS/nixpkgs/pull/171572

File: pkgs/development/python-modules/ipywidgets/default.nix [hunk 5]
```diff
@@ -1,48 +1,40 @@
...
+, pytestCheckHook
 , traitlets
-, notebook
 , widgetsnbextension
 }:
 
 buildPythonPackage rec {
   pname = "ipywidgets";
   version = "7.7.0";
+  format = "setuptools";
```
Comment (FRidh) [inline pkgs/development/python-modules/ipywidgets/default.nix:16]: Why? Can we not switch to pyproject right away?

File: pkgs/development/python-modules/nbconvert/default.nix [hunk 6]
```diff
@@ -1,34 +1,37 @@
...
-, jupyter-client
-, defusedxml
+, traitlets
 }:
 
 buildPythonPackage rec {
   pname = "nbconvert";
-  version = "6.4.5";
+  version = "6.5.0";
+  format = "setuptools";
```
Comment (FRidh) [inline pkgs/development/python-modules/nbconvert/default.nix:30]: can we use pyproject?

File: pkgs/development/python-modules/nbclient/default.nix (hunk 1 above)
Comment (samuela) [reply pkgs/development/python-modules/nbclient/default.nix:30]: this is intentional. GitHub has a nasty bug when you happen to have both a branch and a tag that use the same name. therefore nixpkgs-update and other tools are transitioning to use refs/tags/foo

File: pkgs/development/python-modules/ipywidgets/default.nix (hunk 5 above)
Comment (samuela) [reply pkgs/development/python-modules/ipywidgets/default.nix:16]: Good question... there's a pyproject.toml but it appears to only have linting/typechecking rules in it, so we're stuck with setuptools for the time being unless I'm mistaken.

File: pkgs/development/python-modules/nbconvert/default.nix (hunk 6 above)
Comment (samuela) [reply pkgs/development/python-modules/nbconvert/default.nix:30]: same as https://github.com/NixOS/nixpkgs/pull/171563#discussion_r865310873

File: pkgs/development/python-modules/nbclient/default.nix (hunk 2 above)
Comment (samuela) [reply pkgs/development/python-modules/nbclient/default.nix:22]: maybe but this is already a source of confusion without the comment as evidenced in other reviews: https://github.com/NixOS/nixpkgs/pull/171563#discussion_r865310873 😜

I was also confused by this when I first read the code!

File: pkgs/development/python-modules/nbconvert/default.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (samuela) [reply pkgs/development/python-modules/nbconvert/default.nix:48]: This whole PR should be rebased/redone after https://github.com/NixOS/nixpkgs/pull/171572 and https://github.com/NixOS/nixpkgs/pull/168473 land.

**Comment by samuela** [comment]
closing as stale

**Pull request closed without merging**